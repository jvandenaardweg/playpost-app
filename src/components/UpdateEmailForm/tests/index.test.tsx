import React from 'react';
import { render, RenderAPI, fireEvent } from 'react-native-testing-library';

import { UpdateEmailForm } from '../index';

describe('UpdateEmailForm', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;
    const onChangeTextHandler = jest.fn();
    const onPressUpdateEmailHandler = jest.fn();

    beforeEach(() => {
      wrapper = render(
        <UpdateEmailForm onChangeText={onChangeTextHandler} onPressUpdateEmail={onPressUpdateEmailHandler} email="" isLoading={false} isSuccess={false} />
      );
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    let wrapper: RenderAPI;
    const onChangeTextHandler = jest.fn();
    const onPressUpdateEmailHandler = jest.fn();

    beforeEach(() => {
      wrapper = render(
        <UpdateEmailForm onChangeText={onChangeTextHandler} onPressUpdateEmail={onPressUpdateEmailHandler} email="" isLoading={false} isSuccess={false} />
      );
    });

    it('should fire onChangeText when filling in the reset token TextInput', () => {
      const exampleEmail = 'jordyvandenaardweg@gmail.com';
      const emailTextInput = wrapper.getByTestId('UpdateEmailForm-TextInput-email');

      fireEvent.changeText(emailTextInput, exampleEmail);
      expect(onChangeTextHandler).toHaveBeenCalledWith('email', exampleEmail);
    });

    it('should disable the buttons if isLoading is true', () => {
      wrapper.update(
        <UpdateEmailForm onChangeText={onChangeTextHandler} onPressUpdateEmail={onPressUpdateEmailHandler} email="" isLoading={true} isSuccess={false} />
      );

      expect(wrapper.getByTestId('UpdateEmailForm-Button-update').props.disabled).toBe(true);
    });

    it('should disable the buttons if isSuccess is true', () => {
      wrapper.update(
        <UpdateEmailForm onChangeText={onChangeTextHandler} onPressUpdateEmail={onPressUpdateEmailHandler} email="" isLoading={true} isSuccess={true} />
      );

      expect(wrapper.getByTestId('UpdateEmailForm-Button-update').props.disabled).toBe(true);
    });
  });
});
