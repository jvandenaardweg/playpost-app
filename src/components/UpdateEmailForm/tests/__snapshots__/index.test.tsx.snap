// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateEmailForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="UpdateEmailForm"
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoFocus={false}
      blurOnSubmit={false}
      clearButtonMode="always"
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Your new e-mail address"
      returnKeyType="done"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#e5e5e5",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 15,
          "height": 50,
          "marginBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      testID="UpdateEmailForm-TextInput-email"
      textContentType="emailAddress"
      value=""
    />
    <View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="UpdateEmailForm-Button-update"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000000",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Update e-mail address
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
